Code/Framework/AzFramework/AzFramework/Quality/QualitySystemComponent.cpp

```diff
@@ -84,6 +91,248 @@ namespace AzFramework
+                            }
+                        }
+                        stringValue = AZ::ConsoleTypeHelpers::ToString(value);
+                    }
+                    break;
+                case Type::Boolean:
+                    if (bool value; registry->Get(value, visitArgs.m_jsonKeyPath))
+                    {
+                        stringValue = AZ::ConsoleTypeHelpers::ToString(value);
+                    }
+                    break;
+                case Type::Integer:
+                    if (AZ::s64 value; registry->Get(value, visitArgs.m_jsonKeyPath))
+                    {
+                        // handle Min/Max rule resolution for numbers
+                        if (iter != settingsMap.end() && ruleResolution != DeviceRuleResolution::Last)
+                        {
+                            AZ::s64 currentValue = 0;
+                            AZ::ConsoleTypeHelpers::StringToValue(currentValue, iter->second->value);
+                            if ((ruleResolution == DeviceRuleResolution::Min && currentValue <= value) ||
+                                (ruleResolution == DeviceRuleResolution::Max && currentValue >= value))
+                            {
+                                // disregard this entry because we already have a better value
+                                return AZ::SettingsRegistryInterface::VisitResponse::Skip;
+                            }
+                        }
+                        stringValue = AZ::ConsoleTypeHelpers::ToString(value);
+                    }
+                    break;
+                case Type::String:
+                    {
+                        registry->Get(stringValue, visitArgs.m_jsonKeyPath);
+                    }
+                    break;
+                default:
+                    {
+                        AZ_Warning(
+                            "QualitySystemComponent",
+                            false,
+                            "Skipping settings entry '%.*s' that is not a string, number or bool type.",
+                            AZ_STRING_ARG(visitArgs.m_fieldName));
+                        return AZ::SettingsRegistryInterface::VisitResponse::Skip;
+                    }
+                }
+
+                // if this setting already exists then we are replacing the existing setting value 
+                // and should remove the old one from the ordered settings list
+                if (iter != settingsMap.end())
+                {
+                    settingsList.erase(iter->second);
+                }
+
+                // add the setting to the back of the list
+                settingsList.push_back({ AZStd::string(visitArgs.m_fieldName), stringValue.c_str() });
+
+                // update/store a mapping of "setting" -> list iterator in the settings map
+                settingsMap[visitArgs.m_fieldName] = --settingsList.end();

```

Comment:
It is better to use `emplace(settingsLists.end(), RuleSetting{ AZStd::string(visitArgs.m_fieldName), stringValue.c_str() }` here as that doesn't rely on the list iterator being decrementable from the end.
```suggestion
                newSettingIter = settingsList.emplace(settingsList.end(), RuleSetting{ AZStd::string(visitArgs.m_fieldName), stringValue.c_str() });

                // update/store a mapping of "setting" -> list iterator in the settings map
                settingsMap[visitArgs.m_fieldName] = newSettingIter;

```